design/FY2017/NFP-Single-Zone VAV Fan Control for PackagedTerminalUnits.md

```diff
@@ -0,0 +1,510 @@
+![](FanCoil_ASHRAE901_V86_SPc.png)<br>
+
+# AirloopHVAC:UnitarySystem #
+
+The SZVAV model was then added to the UnitarySystem object in a similar manner to that used for fan coils.
+
+A new choice was added to specify the control type:
+
+    A2,  \field Control Type
+         \type choice
+         \key Load
+         \key SetPoint
+         \key SingleZoneVAV
+         \default Load
+
+Two existing inputs were used to specify the minimum and maximum supply air temperature:
+
+    N2,  \field Minimum Supply Air Temperature
+         \type real
+         \units C
+         \minimum 0.0
+         \maximum 20.0
+         \default 2.0
+         \autosizable
+         \note When Use DOAS DX Cooling Coil is specified as Yes, Minimum Supply Air Temperature
+         \note defines the minimum DOAS DX cooling coil leaving air temperature that should
+         \note be maintained to avoid frost formation. This field is not autosizable when
+         \note the input for Use DOAS DX Cooling Coil = Yes.
+         \note When Control Type = SingleZoneVAV, enter the minimum air temperature limit for reduced fan speed.
+    N17, \field Maximum Supply Air Temperature
+         \type real
+         \units C
+         \autosizable
+         \default 80.0
+         \note Enter the maximum supply air temperature leaving the heating coil.
+         \note When Control Type = SingleZoneVAV, enter the maximum air temperature limit for reduced fan speed.
+
+
+ For this implementation, the SZVAV model was improved to better control the equipment capacity such that how the model performs is more like the description in the Standard. For example, the Standard describes "At cooling demands less than or equal to 50%" where the figure below shows the zone design loads and using this information the model can exactly model 1/2 the design load or any fraction of the design load desired. This improvement to the SZVAV model would have been applied to the fan coil model except that specific code was developed separately for each equipment type (i.e., this improvement was not included for the fan coil model).
+
+![](UnitarySystem_SZVAV_watercoils.png)<br>
+
+
+## Approach ##
+
+The team will apply the previously developed SZVAV modeling methodology to the zone packaged terminal unit objects.
+
+The SZVAV model is now included in fan coil units and unitary systems. Regarding future maintenance it would be better to provide a common function for use by all equipment models. Since the SZVAV model uses specific array variables based on equipment type, a method to pass the necessary information via a function call is warranted. For example, a call to a common routine could pass all necessary information as independent variables:
```

Comment:
If we want to pursue the ultimate right way to do this, it's going to involve extracting parts of the individual data structures into a common base class.  If we can find what parts of each component are common, then we can write functions that accept any variation of this base class and can perform operations easily.  This will reduce the burden of adding `if` blocks inside the functions that are supposed to be generic.  I'd be happy to chat more about that aspect, but I won't just dump it all onto you in this comment.